### DISCUSSION TITLE
Diagnostic traces with UPPAAL 4.1.20-stratego-9

### DISCUSSION THREAD

**Discussion by pacalet** [UPPAALModelChecker/UPPAAL-Meta#77] [Q&A]
I am testing UPPAAL 4.1.20-stratego-9 (rev. 67D95DBCE6B8B4ED), January 2022, macOS version. I cannot find a way to generate a counter-example trace from the verifier in the GUI.

I checked the _Some_ option of the _Engine Options / Diagnostic Trace_ menu, my property is the kind of property for which a counter-example trace makes sense: `A[] a+b <= 6`, and it fails. But I do not see any button or menu item to export a counter-example trace in the Verifier tab.

So I thought that the export to the simulator could be somehow automatic, as explained in the [documentation](https://docs.uppaal.org/gui-reference/menu-bar/options/#diagnostic-trace) but there is nothing like a counter-example trace in the symbolic or concrete simulator.

I checked with `verifyta -f trace -t 0 model.xml ` (same `verifyta` version as the GUI): it generates a trace.

What am I doing wrong in the GUI or where is this button/menu item that I did not find yet?

**Comment by yrke** [comment]
Hi 

If I understand you correctly, 

When running a query with a diagnostic trace (given one is found), you should see a popup like this:
![image](https://user-images.githubusercontent.com/1086483/155279068-31dd05b6-bbc7-41e2-9511-76a029c17dba.png)
When answering yes the found trace will be opened in the Symbolic Simulator tab, you can then switch to the trace and inspect it. You can also export the trace using the "Save" button in the lower left corner. 
![image](https://user-images.githubusercontent.com/1086483/155279288-819c4811-5be7-4da3-a098-8f40a8981b7d.png)

I hope this helps you, 

Regards, Kenneth

**Reply by pacalet** [reply]
I tested under Debian 11.2 (Bullseye) and yes, this popup appears. Then I tested again under macOS (Monterey 12.1) and no, there is no such popup, not even hidden by other windows. This looks like a bug. Shall I open an issue?

**Reply by yrke** [reply]
Yes, that would be very helpful, and thanks for letting us know.
